include/cgimap/api06/changeset_upload/way.hpp

```diff
@@ -55,12 +56,12 @@ class Way : public OSMObject {
                 .str());
       }
 
-      if (m_way_nodes.size() > WAY_MAX_NODES) {
+      if (m_way_nodes.size() > Options::get_instance().get_way_max_nodes()) {
```

Comment:
This is the basis of the PR, replace all constant expressions with a call to the `Options` object to get the correct configuration value.